### PR TITLE
chore: check duplicated packages in CI

### PR DESCRIPTION
**Pull request by LiKang6688**
- Check duplicated packages in CI
- Check duplicated packages in "lint-staged"
- Add yarn-deduplicat as a devDep and a script for it that fixes the issues

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -33,6 +33,9 @@ jobs:
             echo "@qlik-trial:registry=https://npm.pkg.github.com/
             //npm.pkg.github.com/:_authToken=$GITHUB_TOKEN
             //npm.pkg.github.com/:always-auth=true" >> ~/.npmrc
+      - run:
```
Comment (haxxmaxx) [inline .circleci/config.yml:36]: shouldn't you do this after you install? Or am I misunderstanding how deduplicate works?

File: .github/CONTRIBUTION.md [hunk 2]
```diff
@@ -64,6 +64,12 @@ You can skip git commit hooks with:
 
     git commit -m "commit message" --no-verify
 
+## Deduplicate packages
+
+Yarn v1 does not support package deduplication natively. yarn-deduplicate is used to cleans up yarn.lock by removing duplicates:
+
+      npx yarn-deduplicate yarn.lock
+
```
Comment (haxxmaxx) [inline .github/CONTRIBUTION.md:72]: This feels a bit out of context. All the other docs here explain normal things you do when developing. This is more of a maintenance thing. And pretty generic, it just describes what `yarn-deduplicat` does. I think if we want it it in here, then we should at least have a command for it, if that is possible

**Comment by cbt1** [comment]
What happens when the job fails? How would you fix the issue?

You consider adding `yarn-deduplicat` as a devDep and a script for it that fixes the issues. Also consider adding a "lint-staged" check for it on changes to yarn.lock so that you cannot commit a lock file that will fail on CI.

File: .circleci/config.yml (hunk 1 above)
Comment (LiKang6688) [reply .circleci/config.yml:36]: It does not matter.  As `npx yarn-deduplicate` check yarn.lock file and `yarn install --frozen-lockfile` does not change yarn.lock file

File: .github/CONTRIBUTION.md (hunk 2 above)
Comment (LiKang6688) [reply .github/CONTRIBUTION.md:72]: `npx yarn-deduplicate yarn.lock` is the command.

**Comment by LiKang6688** [comment]
> What happens when the job fails? How would you fix the issue?
> 
> You consider adding `yarn-deduplicat` as a devDep and a script for it that fixes the issues. Also consider adding a "lint-staged" check for it on changes to yarn.lock so that you cannot commit a lock file that will fail on CI.

Sure

File: package.json [hunk 3]
```diff
@@ -139,7 +141,8 @@
     "stylis-plugin-rtl-sc": "npm:stylis-plugin-rtl@1.1.0",
     "ts-jest": "29.1.0",
     "typescript": "5.0.4",
-    "yargs": "17.7.2"
+    "yargs": "17.7.2",
+    "yarn-deduplicate": "^6.0.1"
```
Comment (cbt1) [inline package.json:145]: Pin the version and run `yarn` again
```suggestion
    "yarn-deduplicate": "6.0.1"
```

Comment (LiKang6688) [reply package.json]: Thx

File: .github/CONTRIBUTION.md (hunk 2 above)
Comment (haxxmaxx) [reply .github/CONTRIBUTION.md]: ah you changed it, nice

**Review by cbt1 — APPROVED** [review]
LGTM but I would have expected the `yarn.lock` file to be updated a bit more. Like do we not actually have that many duplicate packages in there?

**Comment by LiKang6688** [comment]
> LGTM but I would have expected the `yarn.lock` file to be updated a bit more. Like do we not actually have that many duplicate packages in there?


No, now it is fairly good as I deduplicated the package before here https://github.com/qlik-oss/sn-table/pull/998. 
When we bump versions of packages, we need to manually deduplicate packages like here https://github.com/qlik-oss/sn-table/pull/999

**Pull request merged**